tests/meltano/cli/test_run.py

```diff
@@ -1085,6 +1085,24 @@ def test_run_mapper_config(
             assert result.exit_code == 1
             assert "Error: Block not-a-valid-mapping-name not found" in result.stderr
 
+        # test mapper/mapping name collision detection
+        args = ["run", tap.name, "mapper-mock", target.name]
+        with mock.patch.object(SingerTap, "discover_catalog"), mock.patch.object(
+            SingerTap, "apply_catalog_rules"
+        ), mock.patch(
+            "meltano.core.plugin_invoker.asyncio"
+        ) as asyncio_mock2, mock.patch(
+            "meltano.core.block.parser.ProjectPluginsService",
+            return_value=project_plugins_service,
+        ):
+            asyncio_mock2.create_subprocess_exec = create_subprocess_exec
+            with pytest.raises(
+                Exception,
+                match="block violates set requirements: Expected unique mappings name not the mapper plugin name: mapper-mock",
```

Comment:
To go along with the mapping/mapper name change from above:

```suggestion
                match="block violates set requirements: Expected unique mappings name not the mapper plugin name: mapper-collision",
```